### PR TITLE
Poet engine pr

### PR DESCRIPTION
**Pull request by nick-drozd**

### CODE REVIEW

File: sdk/examples/poet_python/sawtooth_validator/config/path.py [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    (if set) and operating system.
+    """
+    if 'SAWTOOTH_HOME' in os.environ:
+        home_dir = os.environ['SAWTOOTH_HOME']
+        return PathConfig(
+            config_dir=os.path.join(home_dir, 'etc'),
+            log_dir=os.path.join(home_dir, 'logs'),
+            data_dir=os.path.join(home_dir, 'data'),
+            key_dir=os.path.join(home_dir, 'keys'),
+            policy_dir=os.path.join(home_dir, 'policy'))
```
Comment (peterschwarz) [inline sdk/examples/poet_python/sawtooth_validator/config/path.py]: I think you can remove the policy_dir

File: bin/poet-python [hunk 2]
```diff
@@ -29,6 +29,18 @@ sys.path.insert(0, os.path.join(
...
+    'consensus', 'poet', 'core'))
+sys.path.insert(0, os.path.join(
+    os.path.dirname(os.path.dirname(os.path.realpath(__file__))),
+    'consensus', 'poet', 'common'))
+sys.path.insert(0, os.path.join(
+    os.path.dirname(os.path.dirname(os.path.realpath(__file__))),
+    'consensus', 'poet', 'simulator'))
+# sys.path.insert(0, os.path.join(
+#     os.path.dirname(os.path.dirname(os.path.realpath(__file__))),
+#     'validator'))
```
Comment (aludvik) [inline bin/poet-python]: Remove commented out code.

File: bin/poet [hunk 3]
```diff
@@ -40,7 +40,10 @@ sys.path.insert(0, os.path.join(
     'signing'))
 sys.path.insert(0, os.path.join(
     os.path.dirname(os.path.dirname(os.path.realpath(__file__))),
-    'validator'))
+    'sdk', 'examples', 'poet_python'))
+sys.path.insert(0, os.path.join(
+    os.path.dirname(os.path.dirname(os.path.realpath(__file__))),
+    'sdk', 'python'))
```
Comment (aludvik) [inline bin/poet]: Why does this need to be modified?

File: sdk/examples/poet_python/sawtooth_poet_engine/oracle.py [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+from sawtooth_sdk.protobuf.block_pb2 import BlockHeader
+from sawtooth_sdk.protobuf.consensus_pb2 import ConsensusBlock
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+class PoetOracle:
+    def __init__(self, service, path_config, component_endpoint):
+        self._data_dir = path_config.data_dir
```
Comment (peterschwarz) [inline sdk/examples/poet_python/sawtooth_poet_engine/oracle.py]: Pass the paths separately

File: sdk/examples/poet_python/sawtooth_poet_engine/oracle.py [hunk 5]
```diff
@@ -0,0 +1,332 @@
...
+from sawtooth_sdk.protobuf.client_block_pb2 \
+    import ClientBlockGetResponse
+from sawtooth_sdk.protobuf.block_pb2 import BlockHeader
+from sawtooth_sdk.protobuf.consensus_pb2 import ConsensusBlock
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+class PoetOracle:
```
Comment (peterschwarz) [inline sdk/examples/poet_python/sawtooth_poet_engine/oracle.py:48]: Add some doc string to this, to give it some more context.

File: sdk/examples/poet_python/sawtooth_poet_engine/oracle.py [hunk 6]
```diff
@@ -0,0 +1,332 @@
...
+class _BlockStoreProxy:
+    def __init__(self, service, stream):
+        self._service = service
+        self._stream = stream
+
+    @property
+    def chain_head(self):
+        return PoetBlock(self._service.get_chain_head())
+
+    # this needs a component endpoint
```
Comment (aludvik) [inline sdk/examples/poet_python/sawtooth_poet_engine/oracle.py]: does it though?

**Review by peterschwarz — CHANGES_REQUESTED** [review]

File: sdk/examples/poet_python/sawtooth_poet_engine/engine.py [hunk 7]
```diff
@@ -21,15 +21,29 @@
...
+
+        # state variables
+        self._exit = False
+        self._published = False
+        self._building = False
+        self._committing = False
+
+        self._pending_blocks = queue.Queue()
+
+        time.sleep(10)
```
Comment (aludvik) [inline sdk/examples/poet_python/sawtooth_poet_engine/engine.py]: Delete this sleep

File: sdk/examples/poet_python/sawtooth_poet_engine/engine.py [hunk 8]
```diff
@@ -73,29 +112,56 @@ def _cancel_block(self):
...
 
+    def _summarize_block(self):
+        try:
+            return self._service.summarize_block()
+        except (exceptions.InvalidState, exceptions.BlockNotReady) as err:
+            LOGGER.warning(err)
+            return None
+
     def _finalize_block(self):
         time.sleep(1)
```
Comment (aludvik) [inline sdk/examples/poet_python/sawtooth_poet_engine/engine.py]: Delete this sleep

**Review by aludvik — CHANGES_REQUESTED** [review]

**Review by peterschwarz — APPROVED** [review]

**Review by aludvik — APPROVED** [review]

**Pull request merged**